beetsplug/export.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Exports data from beets
+"""
+
+from __future__ import division, absolute_import, print_function
+
+import json
+import codecs
+
+from datetime import datetime, date
+from beets.plugins import BeetsPlugin
+from beets import ui
+from beets import mediafile
+from beetsplug.info import make_key_filter, library_data, tag_data
+
+json.JSONEncoder.default = lambda self, obj: \
+    (obj.isoformat() if isinstance(obj, datetime) or isinstance(obj, date)
+     else None)
+
+
+class ExportPlugin(BeetsPlugin):
+
+    def __init__(self):
+        super(ExportPlugin, self).__init__()
+
+        self.config.add({
+            'default_format': 'json',
+            'json': {
+                # json module formatting options
+                'formatting': {
+                    'ensure_ascii': False,
+                    'indent': 4,
+                    'separators': (',', ': '),
```

Comment:
`sort_keys` is a nice formatting option too (it ensures stable output for `diff` and the like).